packages/valence-macros/src/helpers.rs

```diff
@@ -0,0 +1,61 @@
+use proc_macro::TokenStream;
+use quote::quote;
+use syn::{
+    parse_macro_input, Attribute, AttributeArgs, DataEnum, DeriveInput, GenericArgument,
+    PathArguments, Type, TypePath,
+};
+
+// Check if theres `skip_update` attribute on the field.
+pub(crate) fn has_skip_update_attr(attrs: &[Attribute]) -> bool {
+    attrs.iter().any(|attr| attr.path.is_ident("skip_update"))
+}
+
+pub(crate) fn get_option_inner_type(ty: &Type) -> Option<&Type> {
+    if let Type::Path(TypePath { path, .. }) = ty {
+        if let Some(segment) = path.segments.last() {
+            if segment.ident == "Option" {
+                if let PathArguments::AngleBracketed(args) = &segment.arguments {
+                    if let Some(GenericArgument::Type(inner_type)) = args.args.first() {
```

Comment:
(i'm not really experienced with `syn`) I wonder if something could go wrong on this?  if any the the `if let` checks above fail we return `None`, could it make sense to error instead?

Reply:
i highlighted this line because for `args.first()`, it's not really clear to me if we could be in a situation where `args` has more than one element somehow. I guess `Result<T, E>` would be two types? but not even sure how we can end up there..

Reply:
Regarding the `args.first()`, we are looking into the args of `Option` which only has 1 arg possible, so i don't see a reason this will fail because before that we are making sure we are working with an `Option` type.